src/dnddiceroller/DiceTower.java

```diff
@@ -16,13 +16,45 @@
  */
 public class DiceTower {
   final int PANEL_COUNT = 3;
-  List<Die> dice;
-
+  private List<NumberedDie> numberedDice;
+  private List<Load> loadDice;
+  private int trayValue;
+  private List<Fudge> fudgeDice;
+  
+  
+  //constructor for array list
   public DiceTower() {
-	this.dice = new ArrayList();
+	this.numberedDice = new ArrayList();
+        this.loadDice=new ArrayList();
+        this.fudgeDice=new ArrayList();
+        
+  }
+  
+  
+  
+  //for loop in this
+  public void dropDice(){
+      for(int i=0; i<this.numberedDice.size();i++){
+          this.numberedDice.get(i).roll();
+      }
+          
   }
   
+  //consturctor for variables out of array list function
   public DiceTower(List dice) {
-	this.dice = dice;
+	this.numberedDice = numberedDice;
+        this.loadDice=loadDice;
+        this.fudgeDice=fudgeDice;
```

Comment:
this constructor does not have declared values for `numberedDice` `loadedDice` and `fudgeDice`